File: backend/src/main/java/edu/cit/lugatiman/grossery/entity/GroceryItem.java
package edu.cit.lugatiman.grossery.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDate;

@Entity
@Table(name = "grocery_item")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class GroceryItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "grocery_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "item_name", nullable = false)
    private String itemName;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(name = "unit")
    private String unit;

    @Column(name = "expected_monthly_consumption")
    private Double expectedMonthlyConsumption;

    @Column(name = "expiration_date")
    private LocalDate expirationDate;
}

File: mobile/app/src/main/java/edu/cit/lugatiman/grossery/features/grocery/GroceryItem.kt
package edu.cit.lugatiman.grossery.features.grocery

data class GroceryItem(
    val id: Long? = null,
    val itemName: String,
    val categoryName: String? = null,
    val unit: String?,
    val expectedMonthlyConsumption: Double?,
    val expirationDate: String? = null,
    val receiptPaths: List<String>? = emptyList()
)
